src/curvefit/solvers/solvers.py

```diff
@@ -99,4 +102,39 @@ def fit(self, data, x_init=None, options=None):
                 xs_opt.append(self.solver.x_opt)
             
             self.x_opt = xs_opt[np.argmin(fun_vals)]
-            self.fun_val_opt = np.min(fun_vals)
\ No newline at end of file
+            self.fun_val_opt = np.min(fun_vals)
+
+    def predict(self, **kwargs):
+        return self.solver.predict(self.x_opt, **kwargs)
+
+
+class GaussianMixturesIntegration(Composite):
```

Comment:
@jizezhang How do we set a model instance for the `CoreModel` for the `GaussianMixturesIntegration`? Thinking of how this will be passed into the eventual `ModelRun`

Reply:
so it would be something like 
```
gm_model = GaussianMixtures(...)
gmi = GaussianMixturesIntegration(gm_model)
core_model = CoreModel(...)
base_solver = ScipyOpt()
gmi.set_solver(solver)
gmi.set_model_instance(core_model)
gmi.fit(data)
```
(maybe I should include `ScipyOpt` as a default solver so we dont have to pass it every time)
let me know if this makes sense :)